### PR TITLE
Enable must_exist parameter for update aliases API

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
The parameter `must_exist` in update index aliases API exists in the code base but actually it doesn't work, `IndicesAliasesRequest` doesn't parse the parameter. This PR enables `must_exist` parameter and makes it work when removing aliases by update aliases API.

Another finding is that delete alias API also use the same TransportAction with update aliases API, which is `TransportIndicesAliasesAction`, the current behavior is that if all of the specified aliases are non-existing, it will throw 404 with `aliases_not_found_exception`, but if only some of the specified aliases are non-existing, it removes successfully, the non-existing aliases are ignored. So in order to make update aliases API consistent with the existing behavior and do not import breaking change, when all of the specified aliases are non-existing, no matter `must_exist` is true or false, we also throw 404 with `aliases_not_found_exception`, just like this:

1. Delete an non-existing alias:
Request:
```
POST _aliases
{
   "actions": [
    {
      "remove": {
        "index": "test",
        "aliases": ["testAlias"]
      }
    }
  ]
}
```
Response:
```
{
  "error": {
    "root_cause": [
      {
        "type": "aliases_not_found_exception",
        "reason": "aliases [testAlias] missing",
        "resource.id": "testAlias",
        "resource.type": "aliases"
      }
    ],
    "type": "aliases_not_found_exception",
    "reason": "aliases [testAlias] missing",
    "resource.id": "testAlias",
    "resource.type": "aliases"
  },
  "status": 404
}
```
2. Add an alias:
```
POST test/_alias/testAlias
```
3. Delete the existing alias with an non-existing alias, it executes successfully
```
POST _aliases
{
   "actions": [
    {
      "remove": {
        "index": "test",
        "aliases": ["testAlias", "testAlias1"]
      }
    }
  ]
}
```
4. If `must_exist` is true, deleting an non-existing alias will throw aliases_not_found_exception
```
POST test/_alias/testAlias

POST _aliases
{
   "actions": [
    {
      "remove": {
        "index": "test",
        "aliases": ["testAlias", "testAlias1"],
        "must_exist":true
      }
    }
  ]
}

```

### Related Issues
https://github.com/opensearch-project/OpenSearch/issues/2949

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaobinlong** [comment]
Really appreciate it if anyone can review this PR. This PR has been opened for a long time.

**Comment by reta** [comment]
> Really appreciate it if anyone can review this PR. This PR has been opened for a long time.

Sorry about that @gaobinlong 

> . So in order to make update aliases API consistent with the existing behavior and do not import breaking change, when all of the specified aliases are non-existing, no matter must_exist is true or false, we also throw 404 with aliases_not_found_exception, just like this:

The existing behavior would conform to the case where `must_exist` is not provided at all (is `null`), right? If that is the case, I think we should not return 404 when `must_exist` is explicitly set to `false`.

File: server/src/main/java/org/opensearch/action/admin/indices/alias/IndicesAliasesRequest.java [hunk 1]
```diff
@@ -220,9 +220,11 @@ private static ObjectParser<AliasActions, Void> parser(String name, Supplier<Ali
             ADD_PARSER.declareField(AliasActions::searchRouting, XContentParser::text, SEARCH_ROUTING, ValueType.INT);
             ADD_PARSER.declareField(AliasActions::writeIndex, XContentParser::booleanValue, IS_WRITE_INDEX, ValueType.BOOLEAN);
             ADD_PARSER.declareField(AliasActions::isHidden, XContentParser::booleanValue, IS_HIDDEN, ValueType.BOOLEAN);
-            ADD_PARSER.declareField(AliasActions::mustExist, XContentParser::booleanValue, MUST_EXIST, ValueType.BOOLEAN);
```
Comment (reta) [inline server/src/main/java/org/opensearch/action/admin/indices/alias/IndicesAliasesRequest.java:223]: :+1: seems like it was long lasting bug (Only the remove action supports this parameter.)

File: rest-api-spec/src/main/resources/rest-api-spec/test/indices.update_aliases/40_remove_with_must_exist.yml [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+  - do:
+      catch: /aliases \[testAlias\*\] missing/
+      indices.update_aliases:
+        body:
+          actions:
+            - remove:
+                index: test_index
+                aliases: [ testAlias* ]
+                must_exist: true
+
```
Comment (reta) [inline rest-api-spec/src/main/resources/rest-api-spec/test/indices.update_aliases/40_remove_with_must_exist.yml:36]: @gaobinlong could you please add one test case for empty aliases (here or in `10_basic`), thank you:
```
  - do:
      catch: /\[aliases\] can't be empty/
      indices.update_aliases:
        body:
          actions:
            - remove:
                index: test_index
                aliases: []
                must_exist: true
```

**Review by reta — APPROVED** [review]

File: rest-api-spec/src/main/resources/rest-api-spec/test/indices.update_aliases/40_remove_with_must_exist.yml (hunk 2 above)
Comment (gaobinlong) [reply rest-api-spec/src/main/resources/rest-api-spec/test/indices.update_aliases/40_remove_with_must_exist.yml:36]: I've added that, thanks!

**Comment by gaobinlong** [comment]
> ❕ Gradle check result for [a93447f](https://github.com/opensearch-project/OpenSearch/commit/a93447f41118730d0dc83de70f02d47d632b23ca): [UNSTABLE](https://build.ci.opensearch.org/job/gradle-check/31886/)
> 
> * **TEST FAILURES:**
> 
> ```
>       1 org.opensearch.remotestore.RemoteIndexPrimaryRelocationIT.testPrimaryRelocationWhileIndexing
> ```
> 
> Please review all [flaky tests](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) that succeeded after retry and create an issue if one does not already exist to track the flaky failure.

testPrimaryRelocationWhileIndexing is a flaky test, there's an issue tracking it: https://github.com/opensearch-project/OpenSearch/issues/9191.

**Comment by reta** [comment]
@gaobinlong could you please backport manually to `2.x`? we also need documentation issue here https://github.com/opensearch-project/documentation-website/issues to update Index Aliases (see please https://opensearch.org/docs/latest/im-plugin/index-alias/#add-or-remove-indexes), thank you

**Comment by gaobinlong** [comment]
> @gaobinlong could you please backport manually to `2.x`? we also need documentation issue here https://github.com/opensearch-project/documentation-website/issues to update Index Aliases (see please https://opensearch.org/docs/latest/im-plugin/index-alias/#add-or-remove-indexes), thank you

I've created a document [issue](https://github.com/opensearch-project/documentation-website/issues/6124) yet, and the backport [PR](https://github.com/opensearch-project/OpenSearch/pull/11833) needs to be reviewed, thanks!

**Pull request merged**